setup.py

```diff
@@ -87,15 +87,24 @@ def get_env(build_dir):
 CMAKE_PATH = ENV.get("CMAKE", find_executable("cmake"))
 MAKE_PATH = ENV.get("MAKE", find_executable("make"))
 MAKE_FLAGS = ENV.get("MAKE_FLAGS", "-j %d" % cpu_count()).split()
-EIGEN3_INCLUDE_DIR = ENV.get("EIGEN3_INCLUDE_DIR")  # directory where eigen is saved
-if EIGEN3_INCLUDE_DIR is not None:
-    EIGEN3_INCLUDE_DIR = os.path.abspath(EIGEN3_INCLUDE_DIR)
 HG_PATH = find_executable("hg")
 CC_PATH = ENV.get("CC", find_executable("gcc"))
 CXX_PATH = ENV.get("CXX", find_executable("g++"))
 INSTALL_PREFIX = os.path.join(get_python_lib(), os.pardir, os.pardir, os.pardir)
 PYTHON = sys.executable
 
+# Try to find Eigen
+EIGEN3_INCLUDE_DIR = ENV.get("EIGEN3_INCLUDE_DIR")  # directory where eigen is saved
+# Default to under the build directory if specified
+if EIGEN3_INCLUDE_DIR is None:
+    EIGEN3_INCLUDE_DIR = os.path.join(BUILD_DIR, "eigen")
```

Comment:
Note that `BUILD_DIR` here might not be what's given in `--build_dir` yet, since that's handled only in `run()`.